tools/data_rewriter.py

```diff
@@ -0,0 +1,168 @@
+import random
+import string
+
+import data_builder
+
+
+unknown_location_acronym = 'tbd'
+
+# https://gist.github.com/rogerallen/1583593
+us_state_to_abbrev = {
+    'Alabama': 'AL',
+    'Alaska': 'AK',
+    'American Samoa': 'AS',
+    'Arizona': 'AZ',
+    'Arkansas': 'AR',
+    'California': 'CA',
+    'Colorado': 'CO',
+    'Connecticut': 'CT',
+    'Delaware': 'DE',
+    'District of Columbia': 'DC',
+    'Florida': 'FL',
+    'Georgia': 'GA',
+    'Guam': 'GU',
+    'Hawaii': 'HI',
+    'Idaho': 'ID',
+    'Illinois': 'IL',
+    'Indiana': 'IN',
+    'Iowa': 'IA',
+    'Kansas': 'KS',
+    'Kentucky': 'KY',
+    'Louisiana': 'LA',
+    'Maine': 'ME',
+    'Maryland': 'MD',
+    'Massachusetts': 'MA',
+    'Michigan': 'MI',
+    'Minnesota': 'MN',
+    'Mississippi': 'MS',
+    'Missouri': 'MO',
+    'Montana': 'MT',
+    'Nebraska': 'NE',
+    'Nevada': 'NV',
+    'New Hampshire': 'NH',
+    'New Jersey': 'NJ',
+    'New Mexico': 'NM',
+    'New York': 'NY',
+    'North Carolina': 'NC',
+    'North Dakota': 'ND',
+    'Northern Mariana Islands':'MP',
+    'Ohio': 'OH',
+    'Oklahoma': 'OK',
+    'Oregon': 'OR',
+    'Pennsylvania': 'PA',
+    'Puerto Rico': 'PR',
+    'Rhode Island': 'RI',
+    'South Carolina': 'SC',
+    'South Dakota': 'SD',
+    'Tennessee': 'TN',
+    'Texas': 'TX',
+    'Utah': 'UT',
+    'Vermont': 'VT',
+    'Virgin Islands': 'VI',
+    'Virginia': 'VA',
+    'Washington': 'WA',
+    'Washington DC': 'DC',
+    'West Virginia': 'WV',
+    'Wisconsin': 'WI',
+    'Wyoming': 'WY',
+    'Unknown Location': unknown_location_acronym,
+}
+
+def random_chars(count):
+    return ''.join(random.choice(string.ascii_lowercase) for _ in range(count))
+
+def gen_id(row):
+    state = row['state']
+    state_abbrev = us_state_to_abbrev[state].lower()
+    city = row['city']
+    city_abbrev = city.replace(' ', '').replace('.', '').lower()
+    if state_abbrev == unknown_location_acronym:
+        city_abbrev = unknown_location_acronym
+    if state_abbrev == 'dc':
+        city_abbrev = 'dc'
+
+    # id_line = f'id: {state_abbrev}-{city_abbrev}-{city_index}'
+    return f"{state_abbrev}-{city_abbrev}-{random_chars(4)}"

```

Comment:
I don't think this is foolproof. There's still a possibility it will generate duplicate IDs. I think we would be safer looking at existing state-city-num ids and incrementing

Reply:
This is my attempt at a simple quick fix. Currently PRs are manually generating IDs. But this was a really good comment because it made me implement `validate_ids_unique` which found the following dups.

* mo-kansascity-6
* tx-austin-8
* wa-seattle-16

Reply:
I think as long as validate_ids_unique is there, then the random chars should be fine. I think it'll be a slim chance we have to do anything, but just in case

Reply:
Although actually if you already have the unique ids, incrementing +1 probably won't be a huge addition?